walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/openid4vci/OpenId4VciIssuanceRequest.kt

```diff
@@ -23,19 +43,128 @@ internal class OpenId4VciIssuanceRequest(
     // Attributes describing the Verified ID (eg. name, issuer, logo, background and text colors).
     override val verifiedIdStyle: VerifiedIdStyle,
 
-    val credentialOffer: CredentialOffer,
+    private val credentialOffer: CredentialOffer,
 
-    val credentialMetadata: CredentialMetadata
+    private val credentialMetadata: CredentialMetadata,
+
+    private val credentialConfiguration: CredentialConfiguration,
+
+    private val libraryConfiguration: LibraryConfiguration
 ) : VerifiedIdIssuanceRequest {
+
+    private val requestFormatter = OpenId4VciIssuanceRequestFormatter(libraryConfiguration)
+
     override suspend fun complete(): VerifiedIdResult<VerifiedId> {
-        TODO("Not yet implemented")
+        val result = getResult {
+            val response = formatAndSendIssuanceRequest()
+            mapToVerifiedId(response)
+        }
+        return result
     }
 
     override fun isSatisfied(): Boolean {
-        TODO("Not yet implemented")
+        val validationResult = requirement.validate()
+        return !validationResult.isFailure
     }
 
     override suspend fun cancel(message: String?): VerifiedIdResult<Unit> {
-        TODO("Not yet implemented")
+        return getResult {
+            throw UserCanceledException(
+                message ?: "User Canceled",
+                VerifiedIdExceptions.USER_CANCELED_EXCEPTION.value
+            )
+        }
+    }
+
+    private suspend fun formatAndSendIssuanceRequest(): RawOpenID4VCIResponse {
+        val accessToken =
```

Comment:
do we need to recurse through any grouprequirements? or can we just assume access token? I think we can just assume but what do you think?

Reply:
It is either Auth flow or Pre Auth flow for now. So, we need not recurse. But we can talk about how to handle when new scenario arises with both in credential offer.